File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscustomerpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsCustomerPagedResponseHrefModel } from "./codatdatacontractsdatasetscustomerpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsCustomerPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsCustomerPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsCustomerPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsCustomerPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsCustomerPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsCustomerPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsCustomerPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsCustomerPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsCustomerPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_customerpagedresponselinksmodel.py
import dataclasses
from ..shared import codat_datacontracts_datasets_customerpagedresponsehrefmodel as shared_codat_datacontracts_datasets_customerpagedresponsehrefmodel
from dataclasses_json import dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsCustomerPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_customerpagedresponsehrefmodel.CodatDataContractsDatasetsCustomerPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_customerpagedresponsehrefmodel.CodatDataContractsDatasetsCustomerPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_customerpagedresponsehrefmodel.CodatDataContractsDatasetsCustomerPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_customerpagedresponsehrefmodel.CodatDataContractsDatasetsCustomerPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    